### PR TITLE
refactor(session replay plugin): use session replay sdk to power plugin

### PR DESCRIPTION
**Pull request by kwalker3690**
### Summary

This PR updates the session replay plugin to use the standalone session replay SDK added in https://github.com/amplitude/Amplitude-TypeScript/pull/559

### Checklist

* [x] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-TypeScript/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:  no

### CODE REVIEW

File: packages/plugin-session-replay-browser/test/session-replay.test.ts [hunk 1]
```diff
@@ -1,62 +1,36 @@
-/* eslint-disable jest/expect-expect */
-import * as AnalyticsClientCommon from '@amplitude/analytics-client-common';
+/* eslint-disable @typescript-eslint/no-unused-vars */
```
Comment (kevinpagtakhan) [inline packages/plugin-session-replay-browser/test/session-replay.test.ts:1]: what are the unused vars?

File: packages/plugin-session-replay-browser/package.json [hunk 2]
```diff
@@ -41,6 +41,7 @@
     "@amplitude/analytics-client-common": ">=1 <3",
     "@amplitude/analytics-core": ">=1 <3",
     "@amplitude/analytics-types": ">=1 <3",
+    "@amplitude/session-replay-browser": "0.1.0",
```
Comment (kevinpagtakhan) [inline packages/plugin-session-replay-browser/package.json:44]: are the dependencies above still accurate?

**Review by kevinpagtakhan — APPROVED** [review]

**Review by jackson-amplitude — APPROVED** [review]
LGTM! so much red!

File: packages/plugin-session-replay-browser/test/session-replay.test.ts [hunk 3]
```diff
@@ -1,62 +1,35 @@
-/* eslint-disable jest/expect-expect */
-import * as AnalyticsClientCommon from '@amplitude/analytics-client-common';
 import { CookieStorage, FetchTransport } from '@amplitude/analytics-client-common';
```
Comment (justin-fiedler) [inline packages/plugin-session-replay-browser/test/session-replay.test.ts:1]: Will importing `@amplitude/analytics-client-common` create circular dependencies? Should we mock these instead?

**Review by justin-fiedler — APPROVED** [review]

File: packages/plugin-session-replay-browser/test/session-replay.test.ts (hunk 3 above)
Comment (kwalker3690) [reply packages/plugin-session-replay-browser/test/session-replay.test.ts:1]: The session replay plugin isn't imported anywhere, definitely not to analytics-client-common, so I don't think this should be an issue! But actually I'll remove the use of this anyways, these values aren't used by the plugin anyways

**Pull request merged**